homeassistant/components/todoist/calendar.py

```diff
@@ -564,72 +579,60 @@ def select_best_task(project_tasks: list[TodoistEvent]) -> TodoistEvent:
             ):
                 event = proposed_event
                 continue
-
         return event
 
     async def async_get_events(
-        self, hass: HomeAssistant, start_date: datetime, end_date: datetime
+        self, start_date: datetime, end_date: datetime
     ) -> list[CalendarEvent]:
         """Get all tasks in a specific time frame."""
         if self._id is None:
+            tasks = await self._api.get_tasks()
             project_task_data = [
                 task
-                for task in self._api.state[TASKS]
+                for task in tasks
                 if not self._project_id_whitelist
-                or task[PROJECT_ID] in self._project_id_whitelist
+                or task.project_id in self._project_id_whitelist
             ]
         else:
-            project_data = await hass.async_add_executor_job(
-                self._api.projects.get_data, self._id
-            )
-            project_task_data = project_data[TASKS]
+            project_task_data = await self._api.get_tasks(project_id=self._id)
 
         events = []
         for task in project_task_data:
-            if task["due"] is None:
+            if task.due is None:
                 continue
-            # @NOTE: _parse_due_date always returns the date in UTC time.
-            due_date: datetime | None = _parse_due_date(
-                task["due"], self._api.state["user"]["tz_info"]["hours"]
+            due_date = dt.parse_datetime(
+                task.due.datetime if task.due.datetime else task.due.date
             )
             if not due_date:
                 continue
-            gmt_string = self._api.state["user"]["tz_info"]["gmt_string"]
-            local_midnight = dt.parse_datetime(
-                due_date.strftime(f"%Y-%m-%dT00:00:00{gmt_string}")
-            )
-            if local_midnight is not None:
-                midnight = dt.as_utc(local_midnight)
-            else:
-                midnight = due_date.replace(hour=0, minute=0, second=0, microsecond=0)
-
-            if start_date < due_date < end_date:
+            due_date = dt.as_utc(due_date)
+            if start_date <= due_date < end_date:
```

Comment:
This looks like a logic change? Start and end date should be exclusive https://developers.home-assistant.io/docs/core/entity/calendar#get-events

Reply:
When I did the requested change the tests wouldn't work without this modification and I couldn't figure out exactly why.  Sounds like I might just need to spend the time to rewrite the tests and understand exactly what's going on here.

Reply:
To be clear, the integration appeared to be working correctly w/o this change in the devcontainer, so it was probably a problem with the test itself.

Reply:
I am ok if you want to revet this behavior change and we can merge as is.  Though yes i agree having more tests when performing a large rewrite would be ideal and am happy to help you with that.

Reply:
Are there any other behavior changes that were needed that I should look out for? I didn't see any.

Reply:
No, besides this change the only logic update was making the add reminder function async